custom_components/tuya_local/devices/blitzwoolf_bwsh5-humidifier.yaml

```diff
@@ -0,0 +1,193 @@
+# sample DPS {'updated_at': 1674438037.778799, '1': True, '4': 'health', '5': True, '10': 24, '12': 75, '13': 55, '14': 77, '18': 'c', '19': 'cancel', '20': 0, '21': True, '22': 0, '23': 'level_1', '26': True, '35': True} 
+name: BlitzWoolf BWSH5 Humidifier
+products: 
+  - id: lqzme7o0kzhumdxr
+primary_entity:
+  entity: humidifier
+  class: humidifier
+  dps:
+    - id: 1
+      name: switch   
+      type: boolean
+      mapping:
+        - dps_val: true
+          icon: "mdi:air-humidifier"
+          value: ON
+        - dps_val: false
+          icon: "mdi:air-humidifier-off"
+          value: OFF
+    - id: 5
+      name: led
+      type: boolean
+    - id: 10
+      name: temperature 
+      type: integer
+      readonly: true
+      unit: C
+    - id: 12
+      name: temperature_f
+      type: integer
+      readonly: true
+      hidden: true
+      unit: F
+    - id: 13             
+      name: humidity
+      type: integer
+      optional: true
+      range:
+        min: 35
+        max: 75
+      mapping:
+        - step: 5
+    - id: 14
+      name: current_humidity
+      type: integer
+      readonly: true
+    - id: 18
+      name: temp_unit_convert
+      type: string
+      hidden: true
+secondary_entities:
+  - entity: switch
+    name: "UV Sterilization"
+    icon: "mdi:sun-wireless"
+    dps:
+      - id: 21
+        name: switch
+        type: boolean
+  - entity: switch
+    name: "Warm mist"
+    icon: "mdi:fire"
+    dps:
+      - id: 26
+        name: switch
+        type: boolean
+  - entity: switch
+    name: "Plasma"
+    icon: "mdi:dots-circle"
+    dps:
+      - id: 35
+        name: switch
+        type: boolean        
+  - entity: switch
+    name: "LED"
+    icon: "mdi:led-off"
+    dps:
+      - id: 5
+        name: switch
+        type: boolean
+        mapping:
+          - dps_val: true
+            icon: "mdi:led-on"
+          - dps_val: false
+            icon: "mdi:led-off"
+  - entity: sensor
+    name: Current humidity
+    class: humidity
+    dps:
+      - id: 14
+        type: integer
+        name: sensor
+        class: measurement
+        unit: "%"
+        readonly: true
+  - entity: sensor
+    name: Current temperature
+    class: temperature
+    dps:
+      - id: 10
+        type: integer
+        name: sensor
+        class: measurement
+        readonly: true
+        unit: C
+      - id: 12
+        name: temperature_f
+        type: integer
+        unit: F
+        readonly: true
+        hidden: true
```

Comment:
There's not really any point in adding these as entities then hiding them if you are not using them to switch temperature units.  To do this, you need specific naming on dp 18, but if both are always correct, it is optional, since HA will do unit conversion anyway.

Reply:
I thought device is better recognised if all data points are specified, that is why I left it hidden. Should i remove it?
Temperature in dps 10 is always in C and dps 12 is in F, no matter what is dps 18. That dps seems to only affects temperature on display of device, as I've written in comment.

Reply:
If they are elsewhere in the config (like the unit dp), they are already covered for detection.  Otherwise, yes, including them will help avoid false detections, but there is no need to set them hidden, someone might find it useful to be able to read them as extra attributes.  I generally only make dp hidden that are already visible elsewhere, for example when merging a power switch and mode into hvac_mode in climate devices, or using the a dp for conditional mapping on another dp, when the dp itself is available on another entity.